### PR TITLE
[Merged by Bors] - feat: measurability of the derivative of a function depending on a parameter

### PR DESCRIPTION
**Pull request by sgouezel**
We reuse the main results that were proved for functions not depending on a parameter (but we need to tweak the definition of the set `A` to let it cover all cases, replacing a large inequality by a strict one).

---

There are also name cleanups (when a set is named `A`, the lemma name should be `not_mem_A` and not `not_mem_a`, whatever the naming convention says), and a few prerequisites. I can split these off in other PRs if it makes things easier to review, just tell me.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/FiniteDimension.lean [hunk 1]
```diff
@@ -509,6 +515,32 @@ theorem HasCompactMulSupport.eq_one_or_finiteDimensional {X : Type*} [Topologica
...
+    exact isCompact_of_isClosed_subset (hr.image Cf) isClosed_ball A
+  refine ⟨fun x s ↦ ?_⟩
+  have L : ∀ᶠ n in (atTop : Filter ℕ), s ≤ ‖c‖^n * r := by
+    have : Tendsto (fun n ↦ ‖c‖^n * r) atTop atTop :=
+      Tendsto.atTop_mul_const rpos (tendsto_pow_atTop_atTop_of_one_lt hc)
+    exact Tendsto.eventually_ge_atTop this s
+  rcases L.exists with ⟨n, hn⟩
+  exact isCompact_of_isClosed_subset (M n x) isClosed_ball (closedBall_subset_closedBall hn)
+
+
```
Comment (YaelDillies) [inline Mathlib/Analysis/NormedSpace/FiniteDimension.lean:543]:
```suggestion
```

**Review by YaelDillies — APPROVED** [review]
Looks reasonable!

File: Mathlib/Topology/UniformSpace/Basic.lean [hunk 2]
```diff
@@ -1968,6 +1968,31 @@ theorem continuous_iff'_left [TopologicalSpace β] {f : β → α} :
   continuous_iff_continuousAt.trans <| forall_congr' fun _ => tendsto_nhds_left
 #align uniform.continuous_iff'_left Uniform.continuous_iff'_left
 
+/-- Consider two functions `f` and `g` which coincide on a set `s` and are continuous there.
+Then there is an open neighborhood of `s` on which `f` and `g` are uniformly close. -/
+lemma exists_is_open_mem_uniformity_of_forall_mem_eq
+    [TopologicalSpace β] {r : Set (α × α)} {s : Set β}
+    {f g : β → α} (hf : ∀ x ∈ s, ContinuousAt f x) (hg : ∀ x ∈ s, ContinuousAt g x)
+    (hfg : ∀ x ∈ s, f x = g x) (hr : r ∈ 𝓤 α) :
```
Comment (YaelDillies) [inline Mathlib/Topology/UniformSpace/Basic.lean:1976]:
```suggestion
    (hfg : s.EqOn f g) (hr : r ∈ 𝓤 α) :
```

**Review by YaelDillies — APPROVED** [review]
I'm not finding any golf anymore.

maintainer merge

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**